app/src/main/java/com/T05/krowdtrialz/model/trial/Trial.java

```diff
@@ -14,28 +13,42 @@
  * */
 public abstract class Trial {
     private User experimenter;
-    private Location location;
-    private LocalDateTime dateCreated;
+    private int longitude;
+    private int latitude;
+    private String dateCreated;
 
     public Trial() {}
 
-    public Trial(User user, Location location) {
+    public Trial(User user, int longitude, int latitude) {
         this.experimenter = user;
-        this.location = location;
-        this.dateCreated = LocalDateTime.now();
+        this.longitude = longitude;
+        this.latitude = latitude;
+        LocalDateTime dateTime = LocalDateTime.now();

```

Comment:
Could we instead do this to maintain the contract with LocalDateTime, and increase the cohesion of the class (i.e. not have all these date/time methods:

```
public abstract class Trial {
    private User experimenter;
    private int longitude;
    private int latitude;
    private String dateCreated;

    public Trial() {}

    public Trial(User user, int longitude, int latitude) {
        this.experimenter = user;
        this.longitude = longitude;
        this.latitude = latitude;
        LocalDateTime dateTime = LocalDateTime.now();
        this.dateCreated = dateTime.toString();
    }

    public Trial(User user, int longitude, int latitude, LocalDateTime date) {
        this(user, longitude, latitude);
        this.dateCreated = date.toString();
    }

    @Exclude
    public LocalDateTime getDateCreated() {
        return LocalDateTime.parse(dateCreated);
    }

    public int getLongitude() {
        return longitude;
    }

    public int getLatitude() { return latitude; }

    public User getExperimenter() {
        return experimenter;
    }
}
```

Reply:
@arrooney I just tried implementing this suggestion and after fixing a few minor errors, I ran into an error where dateCreated is always null. I tried using 'private DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");' and 'LocalDateTime dateTime = LocalDateTime.now();
        this.dateCreated = dateTime.format(formatter);' in the constructor but it doesn't seem to work. I agree that the cohesion would be improved if we could get this working though. It may be something for after the part 3 deadline depending how our progress is in the next few days.

Reply:
What about LocaldateTime.now().toString()?

Reply:
But yeah true, we can change later if we want.